supervisor/shared/bluetooth/file_transfer.c

```diff
@@ -433,27 +474,52 @@ STATIC uint8_t _process_mkdir(const uint8_t *raw_buf, size_t command_len) {
         common_hal_bleio_packet_buffer_write(&_transfer_packet_buffer, (const uint8_t *)&response, sizeof(struct mkdir_status), NULL, 0);
         return ANY_COMMAND;
     }
+    // Check to see if USB has already been mounted. If not, then we "eject" from USB until we're done.
+    #if CIRCUITPY_USB && CIRCUITPY_USB_MSC
+    if (storage_usb_enabled() && !usb_msc_lock()) {
+        response.status = STATUS_ERROR_READONLY;
+        common_hal_bleio_packet_buffer_write(&_transfer_packet_buffer, (const uint8_t *)&response, sizeof(struct mkdir_status), NULL, 0);
+        return ANY_COMMAND;
+    }
+    #endif
     // We need to receive another packet to have the full path.
     if (command_len < header_size + command->path_length) {
         return THIS_COMMAND;
     }
     FATFS *fs = &((fs_user_mount_t *)MP_STATE_VM(vfs_mount_table)->obj)->fatfs;
-    char *path = (char *)((uint8_t *)command) + header_size;
-    // TODO: Check that the final character is a `/`
-    path[command->path_length - 1] = '\0';
+    char *path = (char *)command->path;
+    // -1 because fatfs doesn't want a trailing /
+    if (path[command->path_length - 1] == '/') {
+        path[command->path_length - 1] = '\0';
+    } else {
+        path[command->path_length] = '\0';
+    }
```

Comment:
Since this is done twice, at least, you could refactor this into a small routine.

Also, how do you make sure you aren't writing past the end of `path[]`? Is it defined that the buffer is `n` and the permitted path length is `n - 1`?

Reply:
The path string is part of the `current_command` buffer that is `COMMAND_SIZE`. All of the commands verify that the path lengths will fit in `current_command` and adjust for the null termination.